extension.js

```diff
@@ -1,11 +1,47 @@
 const cuMarkdown = require('markdown-it-church-slavonic');
+const vscode = require('vscode');
+var markdownit = require('markdown-it');
+
+require('supports-color');
+
+const fs = require("fs");
+
+/**
+ * @param {vscode.ExtensionContext} context
+ */
+function activate(context) {
+
+    let disposable = vscode.commands.registerCommand('csl.generateXml', function () {  
+        let currentFile =  vscode.window.activeTextEditor.document.uri.fsPath;
+        var xmlRendrer = markdownit().use(cuMarkdown, { renderer: 'xml' });
+        fs.readFile(currentFile, 'utf8', function(err, data) {
+            if (err) throw err;
+            let xml = xmlRendrer.render(data);
+            let newFileName = currentFile.substr(0, currentFile.lastIndexOf('.')) + '.xml';
```

Comment:
Is it possible to open new tab with XML content **without** writing anything to disc, and without giving a particular name to the file? I am a bit worried about filename clash.

Reply:
I think it's possible, I thought it was required. If write it to a disk no conflict will occur, just overriding if one exists.

Reply:
https://stackoverflow.com/questions/41068197/vscode-create-unsaved-file-and-add-content

The StackOverflow answer explains how to do what we want. You can still give it the same "convenient" name. But since file is not saved, the decision to overwrite or to rename will be handled by user.